tests/test_rtdc_feat_anc_plugin.py

```diff
@@ -905,6 +962,183 @@ def test_pf_wrong_length_2():
+        assert feat.attrs.keys()
+        assert b'CLASS' in feat.attrs.keys()
+        assert b'IMAGE_SUBCLASS' in feat.attrs.keys()
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_mask_plugin_data_with_shape_in_info():
+    """Test storage of mask plugin feature attributes with shape info"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_mask_feature()
+    # Mention feature shape in the info
+    info["feature shapes"] = [(80, 250)]
+    PlugInFeature("mask_other", info)
+    with dclab.new_dataset(h5path) as ds:
+        mask_other = ds["mask_other"]
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("mask_other", mask_other)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "mask_other" in events.keys()
+        feat = h5["events"]['mask_other']
+        assert feat.ndim == 3
+        fshape = feat.shape
+        assert fshape[1] > 3 and fshape[2] > 3
+        assert feat.attrs.keys()
+        assert b'CLASS' in feat.attrs.keys()
+        assert b'IMAGE_SUBCLASS' in feat.attrs.keys()
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_mask_plugin_data_without_shape_in_info():
+    """Test storage of mask plugin feature attributes without shape info"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_mask_feature()
+    PlugInFeature("mask_other", info)
+    with dclab.new_dataset(h5path) as ds:
+        mask_other = ds["mask_other"]
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("mask_other", mask_other)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "mask_other" in events.keys()
+        feat = h5["events"]['mask_other']
+        assert feat.ndim == 3
+        fshape = feat.shape
+        assert fshape[1] > 3 and fshape[2] > 3
+        assert feat.attrs.keys()
+        assert b'CLASS' in feat.attrs.keys()
+        assert b'IMAGE_SUBCLASS' in feat.attrs.keys()
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_wrong_plugin_data_shape():
+    """Test storage of wrong data shape of plugin feature"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_non_scalar_feature()
+    PlugInFeature("image_gauss_filter", info)
+    with dclab.new_dataset(h5path) as ds:
+        image_gauss_filter = ds["image_gauss_filter"]
+        im_shape = image_gauss_filter.shape
+        ext_image_gauss_filter = image_gauss_filter.reshape(1, *im_shape)
+    with RTDCWriter(h5path) as hw:
+        with pytest.raises(ValueError, match="Bad shape"):
```

Comment:
Why do you expect a ValueError here? The feature could be a 3D feature, correct? I would expect the original warning message about assuming a certain shape.

Reply:
Actually, the feature shape would be 4D here (added extra dimension line 1136) because I thought of making a limitation to the function that only saves data of shape below 4D.

I am not sure whether it is a good addition or not (please see the condition in [here]( https://github.com/RaghavaAlajangi/dclab/blob/08c8d2097d24f8c1eef7242ed8d27cd283b09300/dclab/rtdc_dataset/writer.py#L240))